src/PersistentFilters.ts

```diff
@@ -131,6 +137,35 @@ export class PersistentFilters {
         this.updateSessions();
     }
 
+    /**
+     * Adds the name of one recently-edited file to the local store and
+     * updates persistent store. The store contains a
+     * maximum number of entries as described by `maxFileHistory`
+     *
+     * If the entry matches a previous entry it is removed from the list
+     * at that position in the stack.
+     *
+     * @param {ZoweDatasetNodeTemplate} template - the template to add
+     */
+    public async addTemplate(template: ZoweDatasetNodeTemplate) {
```

Comment:
Types belong to declaration/definition, not to a JSDoc. Also it is clear from the function nam that the template argument is a template to be added.

```suggestion
     */
    public async addTemplate(template: ZoweDatasetNodeTemplate): ZoweDatasetNodeTemplate {
```